<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="pandoc" />

        <meta name="author" content="Tao Wu" />
        <meta name="author" content="Xue-Song Liu (Corresponding author)" />
    
        <meta name="date" content="2024-04-15" />
    
    <title>Deep learning reveals the metabolic vulnerability of hypoxic tumor cells and the critical function of FLAD1 mediated tumor hypoxia adaption</title>

        <script src="report_main_files/header-attrs-2.22/header-attrs.js"></script>
        <script src="report_main_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link href="report_main_files/bootstrap-3.3.7/css/bootstrap.min.css" rel="stylesheet" />
        <script src="report_main_files/bootstrap-3.3.7/js/bootstrap.min.js"></script>
        <script src="report_main_files/navigation-1.1/tabsets.js"></script>
        <link href="report_main_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
        <link href="report_main_files/readthedown-0.1/readthedown_fonts_embed.css" rel="stylesheet" />
        <script src="report_main_files/readthedown-0.1/readthedown.js"></script>
    
    
        <style type="text/css">code{white-space: pre;}</style>
    <style type="text/css">
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
          background-color: #ffffff;
          color: #a0a0a0;
        }
      pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
      div.sourceCode
        { color: #1f1c1b; background-color: #ffffff; }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      code span { color: #1f1c1b; } /* Normal */
      code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
      code span.an { color: #ca60ca; } /* Annotation */
      code span.at { color: #0057ae; } /* Attribute */
      code span.bn { color: #b08000; } /* BaseN */
      code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
      code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
      code span.ch { color: #924c9d; } /* Char */
      code span.cn { color: #aa5500; } /* Constant */
      code span.co { color: #898887; } /* Comment */
      code span.cv { color: #0095ff; } /* CommentVar */
      code span.do { color: #607880; } /* Documentation */
      code span.dt { color: #0057ae; } /* DataType */
      code span.dv { color: #b08000; } /* DecVal */
      code span.er { color: #bf0303; text-decoration: underline; } /* Error */
      code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
      code span.fl { color: #b08000; } /* Float */
      code span.fu { color: #644a9b; } /* Function */
      code span.im { color: #ff5500; } /* Import */
      code span.in { color: #b08000; } /* Information */
      code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
      code span.op { color: #1f1c1b; } /* Operator */
      code span.ot { color: #006e28; } /* Other */
      code span.pp { color: #006e28; } /* Preprocessor */
      code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
      code span.sc { color: #3daee9; } /* SpecialChar */
      code span.ss { color: #ff5500; } /* SpecialString */
      code span.st { color: #bf0303; } /* String */
      code span.va { color: #0057ae; } /* Variable */
      code span.vs { color: #bf0303; } /* VerbatimString */
      code span.wa { color: #bf0303; } /* Warning */
    </style>
    
    
    <!-- tabsets -->
    <script>
      $(document).ready(function () {
	  window.buildTabsets("toc");
      });
      $(document).ready(function () {
	  $('.tabset-dropdown > .nav-tabs > li').click(function () {
	      $(this).parent().toggleClass('nav-tabs-open')
	  });
      });
    </script>

    <!-- code folding -->
    
    <!-- code download -->
    
    <!-- tabsets dropdown -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
	  display: inline-table;
	  max-height: 500px;
	  min-height: 44px;
	  overflow-y: auto;
	  background: white;
	  border: 1px solid #ddd;
	  border-radius: 4px;
      }
      
      .tabset-dropdown > .nav-tabs > li.active:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
	  content: "&#xe258;";
	  border: none;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs > li.active {
	  display: block;
      }

      .tabset-dropdown > .nav-tabs > li.active a {
  	  padding: 0 15px !important;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
	  border: none;
	  display: inline-block;
	  border-radius: 4px;
	  background-color: transparent;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
	  display: block;
	  float: none;
      }
      
      .tabset-dropdown > .nav-tabs > li {
	  display: none;
	  margin-left: 0 !important;
      }
    </style>
    
</head>

<body class="preload">

   	
         <!-- readthedown start -->   
   <div id="content" data-toggle="wy-nav-shift">
     <nav id="nav-top" role="navigation" aria-label="top navigation">
       <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
     </nav>
         
   
        
      <h1 class="title">Deep learning reveals the metabolic
vulnerability of hypoxic tumor cells and the critical function of FLAD1
mediated tumor hypoxia adaption</h1>
      
         <!-- readthedown authors -->
   <div id="sidebar">
    <h2><a href="#content">Deep learning reveals the metabolic
vulnerability of hypoxic tumor cells and the critical function of FLAD1
mediated tumor hypoxia adaption</a></h2>
    <div id="toc">
      <ul>
      <li><a href="#dependencies"
      id="toc-dependencies">Dependencies</a></li>
      <li><a href="#network-analysis" id="toc-network-analysis">Network
      analysis</a></li>
      <li><a href="#in-silico-perturbation"
      id="toc-in-silico-perturbation">In silico perturbation</a></li>
      </ul>
    </div>
    <div id="postamble" data-toggle="wy-nav-shift" class="status">
                  <p class="author"><span class="glyphicon glyphicon-user"></span> Tao
Wu</p>
                        <p class="author"><span class="glyphicon glyphicon-user"></span> Xue-Song
Liu (Corresponding author)</p>
                        <p class="date"><span class="glyphicon glyphicon-calendar"></span> 2024-04-15</p>
          </div>
   </div>
     

   
      
   
<!-- Don't indent these lines or it will mess pre blocks indentation --> 
<div id="main">
<div id="dependencies" class="section level1">
<h1>Dependencies</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(yardstick)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggprism)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
</div>
<div id="network-analysis" class="section level1">
<h1>Network analysis</h1>
<p>Sample specific GEMs were download in mat format and convert to xml
format by Matlab. Then we applied the R package Met2Graph to extract
enzyme network from GSMs. In this network, enzymes are the nodes
connected by edges represented by metabolites. Two enzymes are connected
if they catalyze two reactions which produce or consume a specific
metabolite. The code for data format convertion can be found in
<code>code/convert_gems.R</code>.</p>
<p>We use Graph2Vec, an algorithm designed to learn vector
representations for whole graphs, to compute the embedding vector for
each sample’s enzyme network. Then we use cosine similarity to calculate
the similarity between pairs of enzyme networks (the code for
calculating graph similarity can be found in
<code>code/graph_sim.ipynb</code>). Based on the Buffa hypoxia score,
samples are divided into hypoxic (score greater than the upper quartile)
and non-hypoxic groups (score less than the lower quartile). We compared
the differences in network similarity between samples with the same
hypoxia status and those with different hypoxia statuses.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/enzyme_net_sim.csv&quot;</span>,<span class="at">data.table =</span> F)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sim<span class="sc">$</span>V1 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sim) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(sim)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/Pancancer hypoxia scores.txt&quot;</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data.table =</span> F)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>hypo_score_summ <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">up_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">4</span>],</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">down_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>,<span class="st">&quot;-&quot;</span>,patient_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_id, tumour_type, Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">score =</span> Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,hypo_score_summ) <span class="sc">%&gt;%</span> </span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hypo_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&lt;</span> down_quan <span class="sc">~</span> <span class="st">&quot;no-hypo&quot;</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&gt;</span> up_quan <span class="sc">~</span> <span class="st">&quot;hypo&quot;</span>,</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(tumour_type)`</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hypo_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>hypo_samples <span class="ot">&lt;-</span> hypo_score<span class="sc">$</span>patient_id[<span class="fu">which</span>(hypo_score<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;hypo&quot;</span>)]</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>no_hypo_samples <span class="ot">&lt;-</span> hypo_score<span class="sc">$</span>patient_id[<span class="fu">which</span>(hypo_score<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;no-hypo&quot;</span>)]</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>hypo_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">substr</span>(<span class="fu">colnames</span>(sim),<span class="dv">1</span>,<span class="dv">12</span>) <span class="sc">%in%</span> hypo_samples)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>no_hypo_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">substr</span>(<span class="fu">colnames</span>(sim),<span class="dv">1</span>,<span class="dv">12</span>) <span class="sc">%in%</span> no_hypo_samples)</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>filter_sim <span class="ot">&lt;-</span> sim[<span class="fu">c</span>(hypo_idx,no_hypo_idx),<span class="fu">c</span>(hypo_idx,no_hypo_idx)]</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>filter_sim<span class="sc">$</span>row_sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(filter_sim)</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>need_ids <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">rownames</span>(filter_sim),<span class="st">&quot;_&quot;</span>,<span class="fu">colnames</span>(filter_sim))</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>filter_sim <span class="ot">&lt;-</span> filter_sim <span class="sc">%&gt;%</span> </span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(row_sample,<span class="fu">everything</span>())</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>filter_sim <span class="ot">&lt;-</span> filter_sim <span class="sc">%&gt;%</span> </span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(filter_sim), <span class="at">names_to =</span> <span class="st">&quot;col_sample&quot;</span>,</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;sim&quot;</span>)</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">ids =</span> <span class="fu">c</span>(hypo_samples,no_hypo_samples),</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;hypo&quot;</span>,<span class="fu">length</span>(hypo_samples)),<span class="fu">rep</span>(<span class="st">&quot;nohypo&quot;</span>,<span class="fu">length</span>(no_hypo_samples)))</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>filter_sim <span class="ot">&lt;-</span> filter_sim <span class="sc">%&gt;%</span> </span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">row_id =</span> <span class="fu">substr</span>(row_sample,<span class="dv">1</span>,<span class="dv">12</span>),</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">col_id =</span> <span class="fu">substr</span>(col_sample,<span class="dv">1</span>,<span class="dv">12</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>            dt <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">row_id =</span> ids,<span class="at">row_type=</span>type)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,dt <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">col_id =</span> ids,<span class="at">col_type=</span>type))</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>filter_sim <span class="ot">&lt;-</span> filter_sim <span class="sc">%&gt;%</span> </span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comb_type =</span> <span class="fu">paste0</span>(row_type,<span class="st">&quot;-&quot;</span>,col_type))</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>cancer_type <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">ids =</span> <span class="fu">c</span>(<span class="fu">unique</span>(filter_sim<span class="sc">$</span>row_id),<span class="fu">unique</span>(filter_sim<span class="sc">$</span>col_id))</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>()</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>cancer_type<span class="sc">$</span>cancer <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">get_cancer_type</span>(cancer_type<span class="sc">$</span>ids)</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>filter_sim <span class="ot">&lt;-</span> filter_sim <span class="sc">%&gt;%</span> </span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>            cancer_type <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">row_id =</span> ids,<span class="at">row_cancer=</span>cancer)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,cancer_type <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">col_id =</span> ids,<span class="at">col_cancer=</span>cancer))</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;igraph&#39;</span></span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:dplyr&#39;:</span></span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     as_data_frame, groups, union</span></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     decompose, spectrum</span></span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     union</span></span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(filter_sim <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">directed =</span> F)</span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">simplify</span>(dt)</span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>dt_unique <span class="ot">&lt;-</span> <span class="fu">as_data_frame</span>(dt) <span class="sc">%&gt;%</span></span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ids =</span> <span class="fu">paste0</span>(from,<span class="st">&quot;_&quot;</span>,to))</span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> filter_sim <span class="sc">%&gt;%</span> </span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">combid =</span> <span class="fu">paste0</span>(row_sample,<span class="st">&quot;_&quot;</span>,col_sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(combid <span class="sc">%in%</span> dt_unique<span class="sc">$</span>ids) <span class="sc">%&gt;%</span> </span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(row_cancer <span class="sc">==</span> col_cancer) <span class="sc">%&gt;%</span> </span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(row_id <span class="sc">!=</span> col_id)</span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> </span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type2 =</span> <span class="fu">ifelse</span>(col_type <span class="sc">!=</span> row_type,<span class="st">&quot;Type2&quot;</span>,<span class="st">&quot;Type1&quot;</span>))</span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(dt,<span class="at">x=</span><span class="st">&quot;type2&quot;</span>,<span class="at">y=</span><span class="st">&quot;sim&quot;</span>,<span class="at">xlab =</span> F,</span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;Similarity&quot;</span>)<span class="sc">+</span></span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label=</span><span class="st">&quot;p.format&quot;</span>)</span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a><span class="fu">facet</span>(p, <span class="at">facet.by =</span> <span class="st">&quot;row_cancer&quot;</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-4-1.png" width="2400" /></p>
<p>We also calculated metrics to measure the importance of gene nodes in
the enzyme network, including degree centrality, which indicates the
number of connections a gene node has with every other gene; betweenness
centrality, quantifying the number of times a gene node appears on the
shortest path between two other nodes; eigenvector centrality which
quantifies a node’s influence in the network based on its connections to
other high-scoring gene nodes and closeness centrality, which calculates
the length of the shortest path between a gene and all other genes in
the network. These metrics were calculated by R package igraph:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">###计算所有样本网络的节点重要性得分</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>all_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;/home/data/sdb/wt/MetGraph/EnzGraphs/&quot;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pattern =</span> <span class="st">&quot;tsv&quot;</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>my.cluster <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">makeCluster</span>(</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="dv">60</span>, </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">&quot;PSOCK&quot;</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> all_files,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">.packages =</span> <span class="fu">c</span>(<span class="st">&quot;igraph&quot;</span>,<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%dopar%</span> {</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;/home/data/sdb/wt/MetGraph/EnzGraphs/&quot;</span>,i)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  sample <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_enzymes_based_graph.tsv&quot;</span>,<span class="st">&quot;&quot;</span>,i)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(tmp, <span class="at">data.table =</span> F)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(from,to)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  dt_g <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">graph_from_data_frame</span>(dt)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Compute the degree centrality for our graph G. </span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  degr_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(dt_g, <span class="at">v =</span> <span class="fu">V</span>(dt_g), <span class="at">mode =</span> <span class="st">&quot;all&quot;</span>, <span class="at">normalized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  norm_degr_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(dt_g, <span class="at">v =</span> <span class="fu">V</span>(dt_g), <span class="at">mode =</span> <span class="st">&quot;all&quot;</span>, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Compute the eigenvector centrality of our network</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  eign_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">eigen_centrality</span>(dt_g, <span class="at">directed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  eign_cent <span class="ot">&lt;-</span> eign_cent<span class="sc">$</span>vector</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Compute the closeness centraility</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  clos_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">closeness</span>(dt_g, <span class="at">normalized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  norm_clos_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">closeness</span>(dt_g, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Compute betweeness centrality</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  betw_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">betweenness</span>(dt_g, <span class="at">directed =</span> <span class="cn">TRUE</span>, <span class="at">normalized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  norm_betw_cent <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">betweenness</span>(dt_g, <span class="at">directed =</span> <span class="cn">TRUE</span>, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  all_centrality <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">vertex =</span> <span class="fu">names</span>(<span class="fu">V</span>(dt_g)),</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>                               <span class="at">degree =</span> degr_cent, </span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>                               <span class="at">norm_degree =</span> norm_degr_cent,</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>                               <span class="at">eigen =</span> eign_cent, </span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>                               <span class="at">closeness =</span> clos_cent,</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>                               <span class="at">norm_closeness =</span> norm_clos_cent,</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>                               <span class="at">betweeness =</span> betw_cent,</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>                               <span class="at">norm_betweeness =</span> norm_betw_cent)</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  all_centrality<span class="sc">$</span>sample <span class="ot">&lt;-</span> sample</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  all_centrality</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>parallel<span class="sc">::</span><span class="fu">stopCluster</span>(<span class="at">cl =</span> my.cluster)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res, <span class="at">file =</span> <span class="st">&quot;/home/data/sdb/wt/all_centrality.rds&quot;</span>)</span></code></pre></div>
<p>We retrieved the gene-wise vector of centrality measures of each
sample sepcific enzyme network and computed the euclidean distance
between these vectors of pairwise samples.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">####基于（共有）节点重要性分布计算相似性</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/all_centrality.rds&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/Pancancer hypoxia scores.txt&quot;</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data.table =</span> F)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>hypo_score_summ <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">up_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">4</span>],</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">down_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>,<span class="st">&quot;-&quot;</span>,patient_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_id, tumour_type, Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">score =</span> Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,hypo_score_summ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hypo_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&lt;</span> down_quan <span class="sc">~</span> <span class="st">&quot;no-hypo&quot;</span>,</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&gt;</span> up_quan <span class="sc">~</span> <span class="st">&quot;hypo&quot;</span>,</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hypo_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hypo_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>hypo_samples <span class="ot">&lt;-</span> hypo_score<span class="sc">$</span>patient_id[<span class="fu">which</span>(hypo_score<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;hypo&quot;</span>)]</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>no_hypo_samples <span class="ot">&lt;-</span> hypo_score<span class="sc">$</span>patient_id[<span class="fu">which</span>(hypo_score<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;no-hypo&quot;</span>)]</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">12</span>) <span class="sc">%in%</span> hypo_score<span class="sc">$</span>patient_id)</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>cal_sim <span class="ot">&lt;-</span> <span class="cf">function</span>(dt, var){</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  dt_met <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(vertex,sample,var) <span class="sc">%&gt;%</span> </span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> sample, <span class="at">values_from =</span> var) <span class="sc">%&gt;%</span> </span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(dt_met) <span class="ot">&lt;-</span> dt_met<span class="sc">$</span>vertex</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  dt_met<span class="sc">$</span>vertex <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>  dt_met <span class="ot">&lt;-</span> <span class="fu">t</span>(dt_met)</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>  dis <span class="ot">&lt;-</span> <span class="fu">dist</span>(dt_met) <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dis)</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>bet_dis <span class="ot">&lt;-</span> <span class="fu">cal_sim</span>(<span class="at">dt =</span> res, <span class="at">var =</span> <span class="st">&quot;norm_betweeness&quot;</span>)</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>clo_dis <span class="ot">&lt;-</span> <span class="fu">cal_sim</span>(<span class="at">dt =</span> res, <span class="at">var =</span> <span class="st">&quot;norm_closeness&quot;</span>)</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>eig_dis <span class="ot">&lt;-</span> <span class="fu">cal_sim</span>(<span class="at">dt =</span> res, <span class="at">var =</span> <span class="st">&quot;eigen&quot;</span>)</span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>deg_dis <span class="ot">&lt;-</span> <span class="fu">cal_sim</span>(<span class="at">dt =</span> res, <span class="at">var =</span> <span class="st">&quot;norm_degree&quot;</span>)</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>sim_res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">bet_dis =</span> bet_dis, <span class="at">clo_dis =</span> clo_dis,</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>                <span class="at">eig_dis =</span> eig_dis, <span class="at">deg_dis =</span> deg_dis)</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sim_res, <span class="at">file =</span> <span class="st">&quot;/home/data/sdb/wt/met_net_dist.rds&quot;</span>)</span></code></pre></div>
<p>The the differences in this network distance between samples with the
same hypoxia status and those with different hypoxia statuses were
calculated.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gc</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            used  (Mb) gc trigger   (Mb)  max used   (Mb)</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Ncells  8911763 476.0   24229170 1294.0  21150860 1129.6</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Vcells 23520092 179.5  162669053 1241.1 203303115 1551.1</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/Pancancer hypoxia scores.txt&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data.table =</span> F)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>hypo_score_summ <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">up_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">4</span>],</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">down_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>,<span class="st">&quot;-&quot;</span>,patient_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_id, tumour_type, Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">score =</span> Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,hypo_score_summ) <span class="sc">%&gt;%</span> </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hypo_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&lt;</span> down_quan <span class="sc">~</span> <span class="st">&quot;no-hypo&quot;</span>,</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&gt;</span> up_quan <span class="sc">~</span> <span class="st">&quot;hypo&quot;</span>,</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(tumour_type)`</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hypo_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hypo_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>hypo_samples <span class="ot">&lt;-</span> hypo_score<span class="sc">$</span>patient_id[<span class="fu">which</span>(hypo_score<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;hypo&quot;</span>)]</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>no_hypo_samples <span class="ot">&lt;-</span> hypo_score<span class="sc">$</span>patient_id[<span class="fu">which</span>(hypo_score<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;no-hypo&quot;</span>)]</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="do">####</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>sim_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/met_net_dist.rds&quot;</span>)</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>bet_dis <span class="ot">&lt;-</span> sim_res<span class="sc">$</span>bet_dis</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>clo_dis <span class="ot">&lt;-</span> sim_res<span class="sc">$</span>clo_dis</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>eig_dis <span class="ot">&lt;-</span> sim_res<span class="sc">$</span>eig_dis</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>deg_dis <span class="ot">&lt;-</span> sim_res<span class="sc">$</span>deg_dis</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>get_sim <span class="ot">&lt;-</span> <span class="cf">function</span>(sim_dt,hypo,nohypo){</span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>  sim_dt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(sim_dt)</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  sim_dt<span class="sc">$</span>row_sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sim_dt)</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  sim_dt <span class="ot">&lt;-</span> sim_dt <span class="sc">%&gt;%</span> </span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(row_sample,<span class="fu">everything</span>())</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>  sim_dt <span class="ot">&lt;-</span> sim_dt <span class="sc">%&gt;%</span> </span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(sim_dt), <span class="at">names_to =</span> <span class="st">&quot;col_sample&quot;</span>,</span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>                        <span class="at">values_to =</span> <span class="st">&quot;sim&quot;</span>)</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">ids =</span> <span class="fu">c</span>(hypo,nohypo),</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;hypo&quot;</span>,<span class="fu">length</span>(hypo)),<span class="fu">rep</span>(<span class="st">&quot;nohypo&quot;</span>,<span class="fu">length</span>(nohypo)))</span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>  sim_dt <span class="ot">&lt;-</span> sim_dt <span class="sc">%&gt;%</span> </span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">row_id =</span> <span class="fu">substr</span>(row_sample,<span class="dv">1</span>,<span class="dv">12</span>),</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>           <span class="at">col_id =</span> <span class="fu">substr</span>(col_sample,<span class="dv">1</span>,<span class="dv">12</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(.,</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>              dt <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">row_id =</span> ids,<span class="at">row_type=</span>type)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(.,dt <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">col_id =</span> ids,<span class="at">col_type=</span>type))</span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>  cancer_type <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">ids =</span> <span class="fu">c</span>(<span class="fu">unique</span>(sim_dt<span class="sc">$</span>row_id),<span class="fu">unique</span>(sim_dt<span class="sc">$</span>col_id))</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> <span class="fu">distinct_all</span>()</span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>  cancer_type<span class="sc">$</span>cancer <span class="ot">&lt;-</span> EasyBioinfo<span class="sc">::</span><span class="fu">get_cancer_type</span>(cancer_type<span class="sc">$</span>ids)</span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>  sim_dt <span class="ot">&lt;-</span> sim_dt <span class="sc">%&gt;%</span> </span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(.,</span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>              cancer_type <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">row_id =</span> ids,<span class="at">row_cancer=</span>cancer)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(.,cancer_type <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">col_id =</span> ids,<span class="at">col_cancer=</span>cancer))</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(sim_dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">directed =</span> F)</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> <span class="fu">simplify</span>(dt)</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>  dt_unique <span class="ot">&lt;-</span> <span class="fu">as_data_frame</span>(dt) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ids =</span> <span class="fu">paste0</span>(from,<span class="st">&quot;_&quot;</span>,to))</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> sim_dt <span class="sc">%&gt;%</span> </span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">combid =</span> <span class="fu">paste0</span>(row_sample,<span class="st">&quot;_&quot;</span>,col_sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(combid <span class="sc">%in%</span> dt_unique<span class="sc">$</span>ids) <span class="sc">%&gt;%</span> </span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(row_cancer <span class="sc">==</span> col_cancer) <span class="sc">%&gt;%</span> </span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(row_id <span class="sc">!=</span> col_id)</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> </span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">type2 =</span> <span class="fu">ifelse</span>(col_type <span class="sc">!=</span> row_type,<span class="st">&quot;Type2&quot;</span>,<span class="st">&quot;Type1&quot;</span>))</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dt)</span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>bet_dt <span class="ot">&lt;-</span> <span class="fu">get_sim</span>(bet_dis, hypo_samples, no_hypo_samples)</span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>clo_dt <span class="ot">&lt;-</span> <span class="fu">get_sim</span>(clo_dis, hypo_samples, no_hypo_samples)</span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>eig_dt <span class="ot">&lt;-</span> <span class="fu">get_sim</span>(eig_dis, hypo_samples, no_hypo_samples)</span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>deg_dt <span class="ot">&lt;-</span> <span class="fu">get_sim</span>(deg_dis, hypo_samples, no_hypo_samples)</span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(row_id)`</span></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(col_id)`</span></span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(bet_dt,<span class="at">x=</span><span class="st">&quot;type2&quot;</span>,<span class="at">y=</span><span class="st">&quot;sim&quot;</span>,<span class="at">xlab =</span> F,</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>               <span class="at">ylab =</span> <span class="st">&quot;Distance&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Betweeness&quot;</span>)<span class="sc">+</span></span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label=</span><span class="st">&quot;p.format&quot;</span>)</span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a><span class="fu">facet</span>(p, <span class="at">facet.by =</span> <span class="st">&quot;row_cancer&quot;</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-7-1.png" width="2400" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(clo_dt,<span class="at">x=</span><span class="st">&quot;type2&quot;</span>,<span class="at">y=</span><span class="st">&quot;sim&quot;</span>,<span class="at">xlab =</span> F,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">ylab =</span> <span class="st">&quot;Distance&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Closeness&quot;</span>)<span class="sc">+</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label=</span><span class="st">&quot;p.format&quot;</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">facet</span>(p, <span class="at">facet.by =</span> <span class="st">&quot;row_cancer&quot;</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-7-2.png" width="2400" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(eig_dt,<span class="at">x=</span><span class="st">&quot;type2&quot;</span>,<span class="at">y=</span><span class="st">&quot;sim&quot;</span>,<span class="at">xlab =</span> F,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">ylab =</span> <span class="st">&quot;Distance&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Eigenvector&quot;</span>)<span class="sc">+</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label=</span><span class="st">&quot;p.format&quot;</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">facet</span>(p, <span class="at">facet.by =</span> <span class="st">&quot;row_cancer&quot;</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-7-3.png" width="2400" /></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggboxplot</span>(deg_dt,<span class="at">x=</span><span class="st">&quot;type2&quot;</span>,<span class="at">y=</span><span class="st">&quot;sim&quot;</span>,<span class="at">xlab =</span> F,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">ylab =</span> <span class="st">&quot;Distance&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Degree&quot;</span>)<span class="sc">+</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">label=</span><span class="st">&quot;p.format&quot;</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">facet</span>(p, <span class="at">facet.by =</span> <span class="st">&quot;row_cancer&quot;</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-7-4.png" width="2400" /></p>
<p>For each gene, we calculated the difference in the previously
mentioned centrality metrics between hypoxic and non-hypoxic tumors.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/all_centrality.rds&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;data/Pancancer hypoxia scores.txt&quot;</span>,<span class="at">data.table =</span> F)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>hypo_score_summ <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">up_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">4</span>],</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">down_quan =</span> <span class="fu">quantile</span>(Buffa_hypoxia_score_intra_tumour_type)[<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>,<span class="st">&quot;-&quot;</span>,patient_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_id, tumour_type, Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">score =</span> Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,hypo_score_summ) <span class="sc">%&gt;%</span> </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hypo_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&lt;</span> down_quan <span class="sc">~</span> <span class="st">&quot;no-hypo&quot;</span>,</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    score <span class="sc">&gt;</span> up_quan <span class="sc">~</span> <span class="st">&quot;hypo&quot;</span>,</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hypo_type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">12</span>) <span class="sc">%in%</span> hypo_score<span class="sc">$</span>patient_id)</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">substr</span>(sample, <span class="dv">1</span>, <span class="dv">12</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>            hypo_score <span class="sc">%&gt;%</span> <span class="fu">select</span>(patient_id,hypo_type))</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>gene_summ <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(vertex) <span class="sc">%&gt;%</span> </span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(hypo_type))) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(vertex <span class="sc">%in%</span> gene_summ<span class="sc">$</span>vertex)</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(res<span class="sc">$</span>vertex)</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="fu">length</span>(all_genes))</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(diff_res)){</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vertex <span class="sc">==</span> all_genes[i])</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>  metrics <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dt)[<span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>]</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>  dt_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">metrics =</span> metrics,</span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="cn">NA</span>,</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">diff =</span> <span class="cn">NA</span></span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(dt_res)){</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">select</span>(dt_res<span class="sc">$</span>metrics[j], hypo_type) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(tmp) <span class="sc">&lt;</span> <span class="dv">2</span>){</span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>      dt_res<span class="sc">$</span>p_value[j] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>      dt_res<span class="sc">$</span>diff[j] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>      dt_tmp <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(<span class="fu">get</span>(dt_res<span class="sc">$</span>metrics[j]) <span class="sc">~</span> hypo_type, <span class="at">data =</span> dt)</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>      dt_res<span class="sc">$</span>p_value[j] <span class="ot">&lt;-</span> dt_tmp<span class="sc">$</span>p.value</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>      dt_res<span class="sc">$</span>diff[j] <span class="ot">&lt;-</span> <span class="fu">median</span>(dt[,dt_res<span class="sc">$</span>metrics[j]][<span class="fu">which</span>(dt<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;hypo&quot;</span>)],<span class="at">na.rm =</span> T) <span class="sc">-</span> <span class="fu">median</span>(dt[,dt_res<span class="sc">$</span>metrics[j]][<span class="fu">which</span>(dt<span class="sc">$</span>hypo_type <span class="sc">==</span> <span class="st">&quot;no-hypo&quot;</span>)],<span class="at">na.rm =</span> T)</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>  dt_res<span class="sc">$</span>padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(dt_res<span class="sc">$</span>p_value, <span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  dt_res<span class="sc">$</span>gene <span class="ot">&lt;-</span> all_genes[i]</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>  diff_res[[i]] <span class="ot">&lt;-</span> dt_res</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Complete &quot;</span>,i,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(diff_res)</span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(diff_res, <span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/hypo_gene_net_diff.rds&quot;</span>)</span></code></pre></div>
<p>We defined a gene as having a significant difference if it had an FDR
value less than 0.05 (Wilcoxon rank-sum test) and the median of the
centrality metric in hypoxic tumor samples was greater than in
non-hypoxic tumor samples. We filtered for genes with at least one
significantly up regulated centrality metric in hypoxic state and used
these genes to perform over representation enrichment analysis of
metabolic pathways.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gc</span>()</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            used  (Mb) gc trigger   (Mb)  max used   (Mb)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Ncells  8913634 476.1   24229170 1294.0  24229170 1294.0</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Vcells 25391465 193.8  126067738  961.9 203303115 1551.1</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/hypo_gene_net_diff.rds&quot;</span>)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>diff_res_sig <span class="ot">&lt;-</span> diff_res <span class="sc">%&gt;%</span> </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(metrics <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;norm_betweeness&quot;</span>,<span class="st">&quot;norm_closeness&quot;</span>,<span class="st">&quot;eigen&quot;</span>,<span class="st">&quot;norm_degree&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sig_counts =</span> <span class="fu">sum</span>(diff <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sig_counts <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="do">##只要有一个显著</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>sig_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(diff_res_sig<span class="sc">$</span>gene)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>trans <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(sig_genes,<span class="st">&quot;symbol&quot;</span>)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Some ID occurs one-to-many match, like &quot;ENSG00000137843, ENSG00000140521, ENSG00000140650&quot;...</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 99.39% genes are mapped to symbol</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>trans <span class="ot">&lt;-</span> trans[<span class="sc">!</span><span class="fu">duplicated</span>(trans<span class="sc">$</span>input_id),]</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>diff_res_sig <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>  diff_res_sig,</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  trans <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene =</span> input_id)</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(gene)`</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pathway,genes)</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(kegg) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;gs_name&quot;</span>,<span class="st">&quot;input_id&quot;</span>)</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>enz_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(<span class="fu">unique</span>(kegg<span class="sc">$</span>input_id), <span class="at">transTo =</span> <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Some ID occurs one-to-many match, like &quot;A2M, AAAS, ABCB11&quot;...</span></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 99.81% genes are mapped to entrezid</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">left_join</span>(kegg,enz_id)</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(input_id)`</span></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in left_join(kegg, enz_id): Detected an unexpected many-to-many relationship between `x` and `y`.</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Row 1 of `x` matches multiple rows in `y`.</span></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Row 51 of `y` matches multiple rows in `x`.</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ If a many-to-many relationship is expected, set `relationship =</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &quot;many-to-many&quot;` to silence this warning.</span></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gs_name,entrezid) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">entrez_gene =</span> entrezid)</span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>enz_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(<span class="fu">unique</span>(diff_res_sig<span class="sc">$</span>symbol), <span class="at">transTo =</span> <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Some ID occurs one-to-many match, like &quot;ABL1, ACAT2, ADA2&quot;...</span></span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 100% genes are mapped to entrezid</span></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>em <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enricher</span>(enz_id<span class="sc">$</span>entrezid, <span class="at">TERM2GENE=</span>kegg,</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pvalueCutoff =</span> <span class="dv">1</span>, <span class="at">qvalueCutoff =</span><span class="dv">1</span>) <span class="do">##use all gene as background</span></span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>em_res <span class="ot">&lt;-</span> em <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a>em_res <span class="ot">&lt;-</span> em_res <span class="sc">%&gt;%</span> </span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; -.+&quot;</span>,<span class="st">&quot;&quot;</span>,ID)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">%in%</span> kegg<span class="sc">$</span>pathway) </span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;enrichplot&#39;</span></span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:ggpubr&#39;:</span></span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     color_palette</span></span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>em <span class="ot">&lt;-</span> <span class="fu">filter</span>(em, ID <span class="sc">%in%</span> em_res<span class="sc">$</span>ID)</span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>em<span class="sc">@</span>result<span class="sc">$</span>Description <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; -.+&quot;</span>,<span class="st">&quot;&quot;</span>,em<span class="sc">@</span>result<span class="sc">$</span>Description)</span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(em, <span class="at">showCategory=</span><span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels=</span><span class="cf">function</span>(x) stringr<span class="sc">::</span><span class="fu">str_wrap</span>(x, <span class="at">width=</span><span class="dv">40</span>))</span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Scale for y is already present.</span></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adding another scale for y, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-9-1.png" width="2400" /></p>
</div>
<div id="in-silico-perturbation" class="section level1">
<h1>In silico perturbation</h1>
<p>The idea is to use the activity of cell death-related gene sets to
divide cells into dying cells and non-dying cells based on single-cell
data, and then fine-tune the GeneFormer, which is a foundation
transformer model pretrained on a large-scale corpus of ~30 million
single cell transcriptomes to enable context-aware predictions in
settings with limited data in network biology, to predict cell dying
state. We then determined the genes whose in silico deletion in hypoxic
cells or non-hypoxic cells from non-dying cell state significantly
shifted the fine- tuned Geneformer cell embeddings towards the dying
cell states:</p>
<p><img
src="https://picgo-wutao.oss-cn-shanghai.aliyuncs.com/undefinedhypoxia_00.png" /></p>
<p>The raw expression counts matrix for lung cancer single-cell data was
downloaded from the corresponding study and processed using the R
package Seurat. To only retain high-quality data, we removed all cells
that have fewer than 250 genes with mapped reads and contain more than
15% of mitochondrial specific reads. Then, we used TCfinder to predict
cancer cells and retained only the cancer cells for downstream
analysis.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>lung <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/lung_cancer/RNA_rawcounts_matrix.rds&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="do">###remove cell with high MT </span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>scRNA_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts=</span>lung)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>scRNA_obj[[<span class="st">&quot;percent_mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(scRNA_obj, </span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">pattern =</span> <span class="st">&quot;^Mt</span><span class="sc">\\</span><span class="st">.|^MT</span><span class="sc">\\</span><span class="st">.|^mt</span><span class="sc">\\</span><span class="st">.|^Mt-|^MT-|^mt-&quot;</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>scRNA_obj <span class="ot">&lt;-</span> <span class="fu">subset</span>(scRNA_obj, </span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">250</span> <span class="sc">&amp;</span> percent_mt <span class="sc">&lt;</span> <span class="dv">15</span>)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="do">##cancer cell pre</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>lung <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(scRNA_obj, <span class="at">assay=</span><span class="st">&quot;RNA&quot;</span>, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> TCfinder<span class="sc">::</span><span class="fu">pathway_score</span>(<span class="at">expr_data =</span> lung,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">normalized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ps) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(lung)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> callr<span class="sc">::</span><span class="fu">r</span>(<span class="at">func =</span> <span class="cf">function</span>(conda_env, ps){</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  reticulate<span class="sc">::</span><span class="fu">use_python</span>(<span class="fu">paste0</span>(conda_env,<span class="st">&quot;/bin/python&quot;</span>))</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  predict_result <span class="ot">&lt;-</span> TCfinder<span class="sc">::</span><span class="fu">predict_cell</span>(<span class="at">path_score =</span> ps)</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(predict_result)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>},<span class="at">args =</span> <span class="fu">list</span>(<span class="at">conda_env =</span> <span class="st">&quot;/home/wt/miniconda3/envs/tcfinder/&quot;</span>, </span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">ps =</span> ps))</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>tumor_cells <span class="ot">&lt;-</span> res<span class="sc">$</span>barcode[<span class="fu">which</span>(res<span class="sc">$</span>cell_type <span class="sc">==</span> <span class="st">&quot;tumor&quot;</span>)] </span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> lung[,tumor_cells]</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tumor,<span class="at">file =</span> <span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell.rds&quot;</span>)</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="do">###test</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>exp <span class="ot">&lt;-</span> <span class="fu">ReadMtx</span>(</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">mtx =</span> <span class="st">&quot;/home/data/sdc/wt/single_cell/raw_counts/cell_research_2020/lung/LC_counts/matrix.mtx&quot;</span>, </span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="st">&quot;/home/data/sdc/wt/single_cell/raw_counts/cell_research_2020/lung/LC_counts/genes.tsv&quot;</span>,</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">cells =</span> <span class="st">&quot;/home/data/sdc/wt/single_cell/raw_counts/cell_research_2020/lung/LC_counts/barcodes.tsv&quot;</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a><span class="do">###remove cell with high MT </span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>scRNA_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts=</span>exp)</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>scRNA_obj[[<span class="st">&quot;percent_mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(scRNA_obj, </span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">pattern =</span> <span class="st">&quot;^Mt</span><span class="sc">\\</span><span class="st">.|^MT</span><span class="sc">\\</span><span class="st">.|^mt</span><span class="sc">\\</span><span class="st">.|^Mt-|^MT-|^mt-&quot;</span>)</span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>scRNA_obj <span class="ot">&lt;-</span> <span class="fu">subset</span>(scRNA_obj, </span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>                    <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">250</span> <span class="sc">&amp;</span> percent_mt <span class="sc">&lt;</span> <span class="dv">15</span>)</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a><span class="do">###find cancer cell</span></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>lung <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(scRNA_obj, <span class="at">assay=</span><span class="st">&quot;RNA&quot;</span>, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> TCfinder<span class="sc">::</span><span class="fu">pathway_score</span>(<span class="at">expr_data =</span> exp,</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>                              <span class="at">normalized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ps) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(exp)</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> callr<span class="sc">::</span><span class="fu">r</span>(<span class="at">func =</span> <span class="cf">function</span>(conda_env, ps){</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>  reticulate<span class="sc">::</span><span class="fu">use_python</span>(<span class="fu">paste0</span>(conda_env,<span class="st">&quot;/bin/python&quot;</span>))</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>  predict_result <span class="ot">&lt;-</span> TCfinder<span class="sc">::</span><span class="fu">predict_cell</span>(<span class="at">path_score =</span> ps)</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(predict_result)</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>},<span class="at">args =</span> <span class="fu">list</span>(<span class="at">conda_env =</span> <span class="st">&quot;/home/wt/miniconda3/envs/tcfinder/&quot;</span>, </span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>              <span class="at">ps =</span> ps))</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>tumor_cells <span class="ot">&lt;-</span> res<span class="sc">$</span>barcode[<span class="fu">which</span>(res<span class="sc">$</span>cell_type <span class="sc">==</span> <span class="st">&quot;tumor&quot;</span>)] </span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> exp[,tumor_cells]</span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tumor,<span class="at">file =</span> <span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell_test.rds&quot;</span>)</span></code></pre></div>
<p>AUCell was used to calculate activity score of hypoxia gene set and
cell death related gene set for each cell. Gaussian mixture model (GMM)
was used to assign cells into high- and low-score group based on cells’
activity scores.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AUCell)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>lung_tumor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell.rds&quot;</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>hypo_pathway <span class="ot">&lt;-</span> fgsea<span class="sc">::</span><span class="fu">gmtPathways</span>(<span class="st">&quot;~/CHPF/exmaple/Hypoxia_geneset.gmt&quot;</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>all_hypo_genes <span class="ot">&lt;-</span> hypo_pathway <span class="sc">%&gt;%</span> <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>all_hypo_genes <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">hypo_sig =</span> all_hypo_genes</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>cells_AUC <span class="ot">&lt;-</span> <span class="fu">AUCell_run</span>(lung_tumor, all_hypo_genes)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>cells_assignment <span class="ot">&lt;-</span> <span class="fu">AUCell_exploreThresholds</span>(cells_AUC,</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">plotHist=</span><span class="cn">TRUE</span>, <span class="at">assign=</span><span class="cn">TRUE</span>)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>auc_score <span class="ot">&lt;-</span> <span class="fu">getAUC</span>(cells_AUC) <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co">#GMM clustering </span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>fit_GMM <span class="ot">&lt;-</span> <span class="fu">Mclust</span>(auc_score[,<span class="dv">1</span>], <span class="at">G =</span> <span class="dv">2</span>)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>Cluster <span class="ot">&lt;-</span> <span class="fu">list</span>(fit_GMM[<span class="st">&quot;classification&quot;</span>])</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>auc_score<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(auc_score)</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>auc_score<span class="sc">$</span>cluster <span class="ot">&lt;-</span> Cluster[[<span class="dv">1</span>]][[<span class="dv">1</span>]]</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(auc_score, <span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/auc_hypo_cluster.rds&quot;</span>)</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="do">###death</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>lung_tumor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell.rds&quot;</span>)</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>pcd_pathway <span class="ot">&lt;-</span> fgsea<span class="sc">::</span><span class="fu">gmtPathways</span>(<span class="st">&quot;data/pcd_pathway.gmt&quot;</span>)</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>all_pcd_genes <span class="ot">&lt;-</span> pcd_pathway <span class="sc">%&gt;%</span> <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() </span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>all_pcd_genes <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">pcd_sig =</span> all_pcd_genes</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>cells_AUC <span class="ot">&lt;-</span> <span class="fu">AUCell_run</span>(lung_tumor, all_pcd_genes)</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>cells_assignment <span class="ot">&lt;-</span> <span class="fu">AUCell_exploreThresholds</span>(cells_AUC,</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">plotHist=</span><span class="cn">TRUE</span>, <span class="at">assign=</span><span class="cn">TRUE</span>)</span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>auc_score <span class="ot">&lt;-</span> <span class="fu">getAUC</span>(cells_AUC) <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a><span class="co">#GMM clustering </span></span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a>fit_GMM <span class="ot">&lt;-</span> <span class="fu">Mclust</span>(auc_score[,<span class="dv">1</span>], <span class="at">G =</span> <span class="dv">2</span>)</span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a>Cluster <span class="ot">&lt;-</span> <span class="fu">list</span>(fit_GMM[<span class="st">&quot;classification&quot;</span>])</span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>auc_score<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(auc_score)</span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>auc_score<span class="sc">$</span>cluster <span class="ot">&lt;-</span> Cluster[[<span class="dv">1</span>]][[<span class="dv">1</span>]]</span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(auc_score, <span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/auc_pcd_cluster.rds&quot;</span>)</span></code></pre></div>
<p>For GeneFormer fine-tuning, the data was saved as <code>loom</code>
format:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>lung_tumor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell.rds&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="do">###id 转化</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(lung_tumor)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(id, <span class="at">transTo =</span> <span class="st">&quot;ens&quot;</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Some ID occurs one-to-many match, like &quot;A2M, A2MP1, AAAS&quot;...</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 44.48% genes are mapped to ensembl</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> ems_id[<span class="sc">!</span><span class="fu">duplicated</span>(ems_id<span class="sc">$</span>input_id),]</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> ems_id[<span class="sc">!</span><span class="fu">duplicated</span>(ems_id<span class="sc">$</span>ensembl),]</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>lung_tumor <span class="ot">&lt;-</span> lung_tumor[ems_id<span class="sc">$</span>input_id,]</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(lung_tumor) <span class="ot">&lt;-</span> ems_id<span class="sc">$</span>ensembl</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>auc_pcd_cluster <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/auc_pcd_cluster.rds&quot;</span>)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>auc_hypo_cluster <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/auc_hypo_cluster.rds&quot;</span>)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(auc_pcd_cluster,<span class="at">x=</span><span class="st">&quot;cluster&quot;</span>,<span class="at">y=</span><span class="st">&quot;pcd_sig&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-13-1.png" width="2400" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(auc_hypo_cluster,<span class="at">x=</span><span class="st">&quot;cluster&quot;</span>,<span class="at">y=</span><span class="st">&quot;hypo_sig&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-13-2.png" width="2400" /></p>
<p>The data from K.H. et al was used as training data, and 30% of the
training data was applied to monitor training process and tune
hyper-parameters.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>auc_pcd_cluster <span class="ot">&lt;-</span> auc_pcd_cluster <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pcd_type =</span> <span class="fu">ifelse</span>(cluster <span class="sc">==</span> <span class="dv">2</span>, <span class="st">&quot;pcd&quot;</span>,<span class="st">&quot;no_pcd&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell, pcd_type)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>auc_hypo_cluster <span class="ot">&lt;-</span> auc_hypo_cluster <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hypo_type =</span> <span class="fu">ifelse</span>(cluster <span class="sc">==</span> <span class="dv">2</span>, <span class="st">&quot;hypo&quot;</span>,<span class="st">&quot;no_hypo&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell, hypo_type)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat,<span class="at">lib.loc =</span> <span class="st">&quot;~/seuratV4/&quot;</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cell =</span> <span class="fu">colnames</span>(lung_tumor)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(.,auc_pcd_cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(.,auc_hypo_cluster)</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>all_pcd_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> lung_tumor)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>meta1 <span class="ot">&lt;-</span> all_pcd_obj<span class="sc">@</span>meta.data</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>meta1<span class="sc">$</span>orig.ident <span class="ot">&lt;-</span> <span class="fu">rownames</span>(meta1)</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>meta1 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  meta1, meta <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">orig.ident =</span> cell)</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">cell_state_hypo =</span> hypo_type,</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">cell_state_pcd =</span> pcd_type) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(meta1) <span class="ot">&lt;-</span> meta1<span class="sc">$</span>orig.ident</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>all_pcd_obj<span class="sc">@</span>meta.data <span class="ot">&lt;-</span> meta1</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a><span class="do">###抽样训练集和测试集</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">202403211</span>)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>pcd_idx <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;Test&quot;</span>,<span class="fu">ncol</span>(all_pcd_obj))</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>pcd_idx[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pcd_idx),<span class="fu">length</span>(pcd_idx)<span class="sc">*</span><span class="fl">0.7</span>,<span class="at">replace =</span> <span class="cn">FALSE</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Train&quot;</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>all_pcd_obj<span class="sc">$</span>train_test <span class="ot">&lt;-</span> pcd_idx</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratDisk)</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveLoom</span>(all_pcd_obj,</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">filename =</span> <span class="st">&quot;/home/data/sdb/wt/pcd_obj.loom&quot;</span>,</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">gene_col_name =</span> <span class="st">&quot;ensembl_id&quot;</span>,</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(loomR)</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>lfile <span class="ot">&lt;-</span> loomR<span class="sc">::</span><span class="fu">connect</span>(<span class="at">filename =</span> <span class="st">&quot;/home/data/sdb/wt/pcd_obj.loom&quot;</span>, </span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mode =</span> <span class="st">&quot;r+&quot;</span>,<span class="at">skip.validate =</span> T)</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>lfile<span class="sc">$</span><span class="fu">add.col.attribute</span>(<span class="fu">list</span>(<span class="at">n_counts =</span> lfile<span class="sc">$</span>col.attrs<span class="sc">$</span>nCount_RNA[]), </span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>                        <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>lfile<span class="sc">$</span><span class="fu">add.col.attribute</span>(<span class="fu">list</span>(<span class="at">individual =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(all_pcd_obj)), </span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>                        <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>lfile[[<span class="st">&quot;row_attrs&quot;</span>]]</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>lfile[[<span class="st">&quot;col_attrs&quot;</span>]]</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>lfile<span class="sc">$</span><span class="fu">close_all</span>()</span></code></pre></div>
<p>The hyperparameter tuning process is implemented by the Ray Tune
framework and used AUC score as evaluation metric. After hyper-parameter
tuning, we set following hyper-parameters: 3.152009e-04 for learning
rate, cosine for learning rate scheduler, 635.01 for warmup steps, 0.279
for weight decay and 24 for batch size. The Fine-tuned model were
independently validated on data sets from Qian, J. et al. The code for
data creation, fine-tuning, training and validation can be found in
<code>script/dataset.ipynb</code>,
<code>script/fine_tuning.ipynb</code>,
<code>script/gene_former.ipynb</code>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/geneformer_preds.csv&quot;</span>,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data.table =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>V1)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">predictions =</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">[|</span><span class="sc">\\</span><span class="st">]&quot;</span>,<span class="st">&quot;&quot;</span>,predictions)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_wider_delim</span>(<span class="at">cols =</span> predictions,<span class="at">delim =</span> <span class="st">&quot;, &quot;</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">names =</span> <span class="fu">c</span>(<span class="st">&quot;pre1&quot;</span>,<span class="st">&quot;pre2&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pre1 =</span> <span class="fu">as.numeric</span>(pre1),</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">pre2 =</span> <span class="fu">as.numeric</span>(pre2))</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(yardstick)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>pre <span class="ot">&lt;-</span> pre <span class="sc">%&gt;%</span> </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">truth =</span> <span class="fu">ifelse</span>(label_ids <span class="sc">==</span> <span class="dv">0</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>),</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">predicted =</span> <span class="fu">ifelse</span>(pred_ids <span class="sc">==</span> <span class="dv">0</span>, <span class="st">&quot;Class1&quot;</span>,<span class="st">&quot;Class2&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Class1 =</span> pre1,</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">Class2 =</span> pre2)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>pre<span class="sc">$</span>truth <span class="ot">&lt;-</span> <span class="fu">factor</span>(pre<span class="sc">$</span>truth)</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>pr <span class="ot">&lt;-</span>  <span class="fu">pr_auc</span>(pre, truth, Class1)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>roc <span class="ot">&lt;-</span>  <span class="fu">roc_auc</span>(pre, truth, Class1)[<span class="st">&quot;.estimate&quot;</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(.,<span class="dv">2</span>)</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">roc_curve</span>(pre, truth, Class1) <span class="sc">%&gt;%</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">-</span> specificity, <span class="at">y =</span> sensitivity)) <span class="sc">+</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>() <span class="sc">+</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="at">ylim =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="fl">0.75</span>, <span class="at">y=</span><span class="fl">0.7</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;ROC-AUC: &quot;</span>,roc),</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">5</span>)</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-15-1.png" width="2400" /></p>
<p>In order to do in silico perturb, we merge the training set and test
set to increase the sample size:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AUCell)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mclust)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>cal_score <span class="ot">&lt;-</span> <span class="cf">function</span>(exp_path,pathway_path){</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  tumor_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(exp_path)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  pathway <span class="ot">&lt;-</span> fgsea<span class="sc">::</span><span class="fu">gmtPathways</span>(pathway_path)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> pathway <span class="sc">%&gt;%</span> <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> <span class="fu">unname</span>() </span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">sig =</span> all_genes</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  cells_AUC <span class="ot">&lt;-</span> <span class="fu">AUCell_run</span>(tumor_exp, all_genes)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  auc_score <span class="ot">&lt;-</span> <span class="fu">getAUC</span>(cells_AUC) <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  fit_GMM <span class="ot">&lt;-</span> <span class="fu">Mclust</span>(auc_score[,<span class="dv">1</span>], <span class="at">G =</span> <span class="dv">2</span>)</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  Cluster <span class="ot">&lt;-</span> <span class="fu">list</span>(fit_GMM[<span class="st">&quot;classification&quot;</span>])</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  auc_score<span class="sc">$</span>cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(auc_score)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  auc_score<span class="sc">$</span>cluster <span class="ot">&lt;-</span> Cluster[[<span class="dv">1</span>]][[<span class="dv">1</span>]]</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(auc_score)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="do">##hypo</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> <span class="fu">cal_score</span>(<span class="at">exp_path =</span> <span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell_test.rds&quot;</span>,</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pathway_path =</span> <span class="st">&quot;~/CHPF/exmaple/Hypoxia_geneset.gmt&quot;</span>)</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(hypo_score,<span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/hypo_score_test.rds&quot;</span>)</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="do">##death</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>death_score <span class="ot">&lt;-</span> <span class="fu">cal_score</span>(<span class="at">exp_path =</span> <span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell_test.rds&quot;</span>,</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>                          <span class="at">pathway_path =</span> <span class="st">&quot;~/hypoxia_target/data/pcd_pathway.gmt&quot;</span>)</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(death_score,<span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/pcd_score_test.rds&quot;</span>)</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="do">##combine train and test data</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>lung_tumor <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/home/data/sdb/wt/lung_cancer/lung_tumor_cell_test.rds&quot;</span>)</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a><span class="do">###id 转化</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(lung_tumor)</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(id, <span class="at">transTo =</span> <span class="st">&quot;ens&quot;</span>)</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> ems_id[<span class="sc">!</span><span class="fu">duplicated</span>(ems_id<span class="sc">$</span>input_id),]</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> ems_id[<span class="sc">!</span><span class="fu">duplicated</span>(ems_id<span class="sc">$</span>ensembl),]</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>lung_tumor <span class="ot">&lt;-</span> lung_tumor[ems_id<span class="sc">$</span>input_id,]</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(lung_tumor) <span class="ot">&lt;-</span> ems_id<span class="sc">$</span>ensembl</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>auc_pcd_cluster <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/pcd_score_test.rds&quot;</span>)</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>auc_hypo_cluster <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/hypo_score_test.rds&quot;</span>)</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>auc_pcd_cluster <span class="ot">&lt;-</span> auc_pcd_cluster <span class="sc">%&gt;%</span> </span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pcd_type =</span> <span class="fu">ifelse</span>(cluster <span class="sc">==</span> <span class="dv">2</span>, <span class="st">&quot;pcd&quot;</span>,<span class="st">&quot;no_pcd&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell, pcd_type)</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>auc_hypo_cluster <span class="ot">&lt;-</span> auc_hypo_cluster <span class="sc">%&gt;%</span> </span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hypo_type =</span> <span class="fu">ifelse</span>(cluster <span class="sc">==</span> <span class="dv">2</span>, <span class="st">&quot;hypo&quot;</span>,<span class="st">&quot;no_hypo&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(cell, hypo_type)</span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>  <span class="at">cell =</span> <span class="fu">colnames</span>(lung_tumor)</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(.,auc_pcd_cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(.,auc_hypo_cluster)</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>all_pcd_obj_test <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> lung_tumor)</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>meta1 <span class="ot">&lt;-</span> all_pcd_obj_test<span class="sc">@</span>meta.data</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>meta1<span class="sc">$</span>orig.ident <span class="ot">&lt;-</span> <span class="fu">rownames</span>(meta1)</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>meta1 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>  meta1, meta <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">orig.ident =</span> cell)</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">cell_state_hypo =</span> hypo_type,</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>             <span class="at">cell_state_pcd =</span> pcd_type) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(meta1) <span class="ot">&lt;-</span> meta1<span class="sc">$</span>orig.ident</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>all_pcd_obj_test<span class="sc">@</span>meta.data <span class="ot">&lt;-</span> meta1</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> <span class="fu">merge</span>(all_pcd_obj,all_pcd_obj_test)</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratDisk)</span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a><span class="fu">SaveLoom</span>(combined,</span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">filename =</span> <span class="st">&quot;/home/data/sdb/wt/pcd_obj_comb.loom&quot;</span>,</span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a>         <span class="at">gene_col_name =</span> <span class="st">&quot;ensembl_id&quot;</span>,</span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>         <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a>lfile <span class="ot">&lt;-</span> loomR<span class="sc">::</span><span class="fu">connect</span>(<span class="at">filename =</span> <span class="st">&quot;/home/data/sdb/wt/pcd_obj_comb.loom&quot;</span>, </span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mode =</span> <span class="st">&quot;r+&quot;</span>,<span class="at">skip.validate =</span> T)</span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>lfile<span class="sc">$</span><span class="fu">add.col.attribute</span>(<span class="fu">list</span>(<span class="at">n_counts =</span> lfile<span class="sc">$</span>col.attrs<span class="sc">$</span>nCount_RNA[]), </span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>                        <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>lfile<span class="sc">$</span><span class="fu">add.col.attribute</span>(<span class="fu">list</span>(<span class="at">individual =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(combined)), </span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a>                        <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a>lfile[[<span class="st">&quot;row_attrs&quot;</span>]]</span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a>lfile[[<span class="st">&quot;col_attrs&quot;</span>]]</span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>lfile<span class="sc">$</span><span class="fu">close_all</span>()</span></code></pre></div>
<p>In silico perturbation was achieved by removing the given gene from
the rank value encoding of the given single-cell transcriptome and
quantifying the cosine similarity between the original and perturbed
cell embeddings to determine the predicted deleterious impact of
deleting that gene in that cell. This impact was compared with the
random distribution drawn from the other genes to calculate p value and
corresponding FDR. The code for perturbation can be found in
<code>script/in_silico_perturbation.ipynb</code>.</p>
<p>We first performed in silico perturbation on all single cells,
identified genes whose deletion could significantly shift cells from the
non-dying state to the dying state, and compared these genes with the
dependent genes obtained by CRIPSR.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/all_pcd_res_all_cell.csv&quot;</span>,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data.table =</span> F)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">ifelse</span>(Shift_to_goal_end <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> Goal_end_FDR <span class="sc">&lt;</span> <span class="fl">0.0001</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;Hit&quot;</span>,<span class="st">&quot;Not-Hit&quot;</span>))</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>cell_info <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;/home/data/sdb/wt/Model_2023Q4.csv&quot;</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>cell_info <span class="ot">&lt;-</span> cell_info <span class="sc">%&gt;%</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((OncotreeLineage <span class="sc">!=</span> <span class="st">&quot;Normal&quot;</span>) <span class="sc">&amp;</span> (OncotreePrimaryDisease <span class="sc">!=</span> <span class="st">&quot;Non-Cancerous&quot;</span>))</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>lung_cell <span class="ot">&lt;-</span> cell_info <span class="sc">%&gt;%</span> <span class="fu">filter</span>(OncotreeLineage <span class="sc">==</span> <span class="st">&quot;Lung&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(OncotreeSubtype <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Lung Adenocarcinoma&quot;</span>,<span class="st">&quot;Non-Small Cell Lung Cancer&quot;</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;Lung Squamous Cell Carcinoma&quot;</span>,</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;Lung Adenosquamous Carcinoma&quot;</span>))</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>dep_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/CRISPRGeneDependency_2023Q4.csv&quot;</span>,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>                            <span class="at">data.table =</span> F)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(dep_dt)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(dep_dt)] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">([^</span><span class="sc">\\</span><span class="st">)]+</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">colnames</span>(dep_dt)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(dep_dt)])</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>dep_dt <span class="ot">&lt;-</span> dep_dt <span class="sc">%&gt;%</span> </span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(dep_dt)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(dep_dt)],<span class="at">names_to =</span> <span class="st">&quot;gene&quot;</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values_to =</span> <span class="st">&quot;score&quot;</span>)</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>lung_dep <span class="ot">&lt;-</span> dep_dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(V1 <span class="sc">%in%</span> lung_cell<span class="sc">$</span>ModelID) <span class="sc">%&gt;%</span> </span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(score))</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> lung_dep <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> res<span class="sc">$</span>Gene_name)</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(dt, res <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene =</span> Gene_name) <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene, type))</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(gene)`</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>dt_summ <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">pos_cell =</span> <span class="fu">sum</span>(score <span class="sc">&gt;</span> <span class="fl">0.8</span>),</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">type =</span> <span class="fu">unique</span>(type)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>dt_summ <span class="ot">&lt;-</span> dt_summ <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">depmap_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>  pos_cell <span class="sc">&gt;</span> <span class="dv">20</span> <span class="sc">~</span> <span class="st">&quot;yes&quot;</span>,</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>  <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;no&quot;</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggprism)</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> dt_summ <span class="sc">%&gt;%</span> </span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(depmap_type,type) <span class="sc">%&gt;%</span> </span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">gene_counts =</span> <span class="fu">length</span>(<span class="fu">unique</span>(gene))) <span class="sc">%&gt;%</span> </span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">ifelse</span>(type <span class="sc">==</span> <span class="st">&quot;Hit&quot;</span>,<span class="st">&quot;In Silico Perturbation Hit&quot;</span>,</span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;In Silico Perturbation Not Hit&quot;</span>))</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;depmap_type&#39;. You can override using the</span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `.groups` argument.</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dt_summ<span class="sc">$</span>type,dt_summ<span class="sc">$</span>depmap_type) <span class="sc">%&gt;%</span> <span class="fu">chisq.test</span>()</span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction</span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  .</span></span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; X-squared = 731.72, df = 1, p-value &lt; 2.2e-16</span></span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>df,<span class="fu">aes</span>(<span class="at">x=</span>depmap_type,<span class="at">y=</span>gene_counts,<span class="at">fill=</span>type))<span class="sc">+</span></span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>,<span class="at">position=</span><span class="st">&quot;fill&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Percent of cases (%)&quot;</span>,<span class="at">title =</span> <span class="st">&quot;Chi-squared test, P &lt; 2.2e-16&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#FE0000&quot;</span>,<span class="st">&quot;#00FDFE&quot;</span>))<span class="sc">+</span></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;CRIPSR Not Hit&quot;</span>,<span class="st">&quot;CRIPSR Hit&quot;</span>))<span class="sc">+</span></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-17-1.png" width="2400" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>dt_summ, <span class="fu">aes</span>(<span class="at">x=</span>pos_cell, <span class="at">group=</span>type, <span class="at">fill=</span>type)) <span class="sc">+</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">adjust=</span><span class="fl">1.5</span>, <span class="at">alpha=</span>.<span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Counts of CRPSPR Positive Cell&quot;</span>,<span class="at">y=</span><span class="st">&quot;Density&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-17-2.png" width="2400" /></p>
<p>We filtered the results of in silico perturbation by the following
criteria: 1. Only retain metabolism-related genes (KEGG)，2. The effect
value obtained by delete the gene under hypoxia state was &gt; 0, and
the FDR was &lt; 0.05 (the effect value refers to the magnitude of cell
embedding shift from the non-dying state to the dying state after delete
the specific gene), 3. Fold change (absolute value) of the effect size
between hypoxic and non-hypoxic state was &gt; 1, 4. Based on TCGA data,
fold change of mRNA expression between hypoxic and non-hypoxic tumors
was &gt; 1 and corresponding FDR was &lt; 0.05.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>hypo_res <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/0331_pcd_hypo_res.csv&quot;</span>,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">data.table =</span> F)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>no_hypo_res <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/hypoxia_target/data/0331pcd_no_hypo_res.csv&quot;</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">data.table =</span> F)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>all_res <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  hypo_res <span class="sc">%&gt;%</span> <span class="fu">select</span>(Gene_name,Shift_to_goal_end,Goal_end_vs_random_pval,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>                      Goal_end_FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">nchar</span>(Gene_name) <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  no_hypo_res <span class="sc">%&gt;%</span> <span class="fu">select</span>(Gene_name,Shift_to_goal_end) <span class="sc">%&gt;%</span> </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">Shift_no_hypo =</span> Shift_to_goal_end) <span class="sc">%&gt;%</span> </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">nchar</span>(Gene_name) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fc =</span> Shift_to_goal_end<span class="sc">/</span>Shift_no_hypo)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>kegg_all_pathway <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>kegg_all_pathway <span class="ot">&lt;-</span> kegg_all_pathway <span class="sc">%&gt;%</span> </span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>kegg_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(kegg_all_pathway<span class="sc">$</span>genes)</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> all_res <span class="sc">%&gt;%</span> </span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Gene_name <span class="sc">%in%</span> kegg_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Shift_to_goal_end <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> Goal_end_FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">abs_fc =</span> <span class="fu">abs</span>(fc)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Goal_end_FDR, <span class="fu">desc</span>(abs_fc)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stat=</span> (<span class="sc">-</span><span class="fu">log10</span>(Goal_end_FDR)) <span class="sc">*</span> abs_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(stat))</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_res_meta,<span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/all_res_meta0331.rds&quot;</span>)</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a><span class="do">#####分析这些基因的表达</span></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(abs_fc <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>ems_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(all_res_meta<span class="sc">$</span>Gene_name, <span class="at">transTo =</span> <span class="st">&quot;ens&quot;</span>)</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>  all_res_meta,</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>  ems_id <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Gene_name =</span> input_id)</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>mapping <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/gencode.v22.annotation.gene.probeMap&quot;</span>,<span class="at">data.table =</span> F)</span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a>mapping <span class="ot">&lt;-</span> mapping <span class="sc">%&gt;%</span> </span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">gsub</span>(<span class="st">&quot;[.].+&quot;</span>,<span class="st">&quot;&quot;</span>,id) <span class="sc">%in%</span> ems_id<span class="sc">$</span>ensembl)</span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a>  all_res_meta,</span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a>  mapping <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene,id) <span class="sc">%&gt;%</span> </span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">id2 =</span> <span class="fu">gsub</span>(<span class="st">&quot;[.].+&quot;</span>,<span class="st">&quot;&quot;</span>,id)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">ensembl =</span> id2)</span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(id))</span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true" tabindex="-1"></a>dup_gene <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb20-51"><a href="#cb20-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">counts =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(counts <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb20-52"><a href="#cb20-52" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb20-53"><a href="#cb20-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(Gene_name <span class="sc">%in%</span> dup_gene<span class="sc">$</span>Gene_name) <span class="sc">|</span> ((Gene_name <span class="sc">%in%</span> dup_gene<span class="sc">$</span>Gene_name) <span class="sc">&amp;</span> (Gene_name <span class="sc">==</span> gene)))</span>
<span id="cb20-54"><a href="#cb20-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-55"><a href="#cb20-55" aria-hidden="true" tabindex="-1"></a>luad <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/TCGA-LUAD.htseq_fpkm-uq.tsv.gz&quot;</span>,</span>
<span id="cb20-56"><a href="#cb20-56" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data.table =</span> F)</span>
<span id="cb20-57"><a href="#cb20-57" aria-hidden="true" tabindex="-1"></a>luad <span class="ot">&lt;-</span> luad <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> all_res_meta<span class="sc">$</span>id)</span>
<span id="cb20-58"><a href="#cb20-58" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(luad) <span class="ot">&lt;-</span> luad<span class="sc">$</span>Ensembl_ID</span>
<span id="cb20-59"><a href="#cb20-59" aria-hidden="true" tabindex="-1"></a>luad<span class="sc">$</span>Ensembl_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb20-60"><a href="#cb20-60" aria-hidden="true" tabindex="-1"></a>lusc <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/TCGA-LUSC.htseq_fpkm-uq.tsv.gz&quot;</span>,</span>
<span id="cb20-61"><a href="#cb20-61" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data.table =</span> F)</span>
<span id="cb20-62"><a href="#cb20-62" aria-hidden="true" tabindex="-1"></a>lusc <span class="ot">&lt;-</span> lusc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> all_res_meta<span class="sc">$</span>id)</span>
<span id="cb20-63"><a href="#cb20-63" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(lusc) <span class="ot">&lt;-</span> lusc<span class="sc">$</span>Ensembl_ID</span>
<span id="cb20-64"><a href="#cb20-64" aria-hidden="true" tabindex="-1"></a>lusc<span class="sc">$</span>Ensembl_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb20-65"><a href="#cb20-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-66"><a href="#cb20-66" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">colnames</span>(luad),<span class="fu">colnames</span>(lusc))</span>
<span id="cb20-67"><a href="#cb20-67" aria-hidden="true" tabindex="-1"></a>all_cancers <span class="ot">&lt;-</span> all_samples[<span class="fu">as.numeric</span>(<span class="fu">substr</span>(all_samples,<span class="dv">14</span>,<span class="dv">15</span>)) <span class="sc">&lt;</span> <span class="dv">11</span>] <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb20-68"><a href="#cb20-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-69"><a href="#cb20-69" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(luad,lusc)</span>
<span id="cb20-70"><a href="#cb20-70" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> all_cancer_exp[,all_cancers]</span>
<span id="cb20-71"><a href="#cb20-71" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> <span class="fu">t</span>(all_cancer_exp) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb20-72"><a href="#cb20-72" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> <span class="fu">apply</span>(all_cancer_exp,<span class="dv">2</span>,<span class="cf">function</span>(x){(<span class="dv">2</span><span class="sc">^</span>x)<span class="sc">-</span><span class="dv">1</span>}) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb20-73"><a href="#cb20-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-74"><a href="#cb20-74" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;data/Pancancer hypoxia scores.txt&quot;</span>,<span class="at">data.table =</span> F)</span>
<span id="cb20-75"><a href="#cb20-75" aria-hidden="true" tabindex="-1"></a>hypo_score <span class="ot">&lt;-</span> hypo_score <span class="sc">%&gt;%</span> </span>
<span id="cb20-76"><a href="#cb20-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">patient_id =</span> <span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>,<span class="st">&quot;-&quot;</span>,patient_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-77"><a href="#cb20-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(tumour_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;LUAD&quot;</span>,<span class="st">&quot;LUSC&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-78"><a href="#cb20-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(patient_id, tumour_type, Buffa_hypoxia_score_intra_tumour_type) <span class="sc">%&gt;%</span> </span>
<span id="cb20-79"><a href="#cb20-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb20-80"><a href="#cb20-80" aria-hidden="true" tabindex="-1"></a>    tumour_type <span class="sc">==</span> <span class="st">&quot;LUAD&quot;</span> <span class="sc">&amp;</span> Buffa_hypoxia_score_intra_tumour_type <span class="sc">&gt;</span> <span class="dv">21</span> <span class="sc">~</span> <span class="st">&quot;hypo&quot;</span>,</span>
<span id="cb20-81"><a href="#cb20-81" aria-hidden="true" tabindex="-1"></a>    tumour_type <span class="sc">==</span> <span class="st">&quot;LUAD&quot;</span> <span class="sc">&amp;</span> Buffa_hypoxia_score_intra_tumour_type <span class="sc">&lt;</span> (<span class="sc">-</span><span class="dv">21</span>) <span class="sc">~</span> <span class="st">&quot;normal&quot;</span>,</span>
<span id="cb20-82"><a href="#cb20-82" aria-hidden="true" tabindex="-1"></a>    tumour_type <span class="sc">==</span> <span class="st">&quot;LUSC&quot;</span> <span class="sc">&amp;</span> Buffa_hypoxia_score_intra_tumour_type <span class="sc">&gt;</span> <span class="dv">15</span> <span class="sc">~</span> <span class="st">&quot;hypo&quot;</span>,</span>
<span id="cb20-83"><a href="#cb20-83" aria-hidden="true" tabindex="-1"></a>    tumour_type <span class="sc">==</span> <span class="st">&quot;LUSC&quot;</span> <span class="sc">&amp;</span> Buffa_hypoxia_score_intra_tumour_type <span class="sc">&lt;</span> (<span class="sc">-</span><span class="dv">13</span>) <span class="sc">~</span> <span class="st">&quot;normal&quot;</span>,</span>
<span id="cb20-84"><a href="#cb20-84" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb20-85"><a href="#cb20-85" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb20-86"><a href="#cb20-86" aria-hidden="true" tabindex="-1"></a>all_cancer_exp<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(all_cancer_exp),<span class="dv">1</span>,<span class="dv">12</span>)</span>
<span id="cb20-87"><a href="#cb20-87" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb20-88"><a href="#cb20-88" aria-hidden="true" tabindex="-1"></a>  all_cancer_exp,</span>
<span id="cb20-89"><a href="#cb20-89" aria-hidden="true" tabindex="-1"></a>  hypo_score <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sample =</span> patient_id) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,type)</span>
<span id="cb20-90"><a href="#cb20-90" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-91"><a href="#cb20-91" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> all_cancer_exp <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,type,<span class="fu">everything</span>())</span>
<span id="cb20-92"><a href="#cb20-92" aria-hidden="true" tabindex="-1"></a>all_cancer_exp <span class="ot">&lt;-</span> all_cancer_exp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(type <span class="sc">!=</span> <span class="st">&quot;others&quot;</span>)</span>
<span id="cb20-93"><a href="#cb20-93" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ids =</span> <span class="fu">colnames</span>(all_cancer_exp)[<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(all_cancer_exp)],</span>
<span id="cb20-94"><a href="#cb20-94" aria-hidden="true" tabindex="-1"></a>                       <span class="at">exp_fc =</span> <span class="cn">NA</span>, <span class="at">p_value =</span> <span class="cn">NA</span>)</span>
<span id="cb20-95"><a href="#cb20-95" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(diff_res)){</span>
<span id="cb20-96"><a href="#cb20-96" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> all_cancer_exp <span class="sc">%&gt;%</span> </span>
<span id="cb20-97"><a href="#cb20-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(type,diff_res<span class="sc">$</span>ids[i]) <span class="sc">%&gt;%</span> </span>
<span id="cb20-98"><a href="#cb20-98" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">exp =</span> <span class="dv">2</span>)</span>
<span id="cb20-99"><a href="#cb20-99" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;hypo&quot;</span>,<span class="st">&quot;normal&quot;</span>))</span>
<span id="cb20-100"><a href="#cb20-100" aria-hidden="true" tabindex="-1"></a>  dt_res <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(exp <span class="sc">~</span> type, <span class="at">data=</span>dt)</span>
<span id="cb20-101"><a href="#cb20-101" aria-hidden="true" tabindex="-1"></a>  diff_res<span class="sc">$</span>exp_fc[i] <span class="ot">&lt;-</span> <span class="fu">mean</span>(dt<span class="sc">$</span>exp[dt<span class="sc">$</span>type <span class="sc">==</span> <span class="st">&quot;hypo&quot;</span>]) <span class="sc">/</span> <span class="fu">mean</span>(dt<span class="sc">$</span>exp[dt<span class="sc">$</span>type <span class="sc">==</span> <span class="st">&quot;normal&quot;</span>])</span>
<span id="cb20-102"><a href="#cb20-102" aria-hidden="true" tabindex="-1"></a>  diff_res<span class="sc">$</span>p_value[i] <span class="ot">&lt;-</span> dt_res<span class="sc">$</span>p.value</span>
<span id="cb20-103"><a href="#cb20-103" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-104"><a href="#cb20-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-105"><a href="#cb20-105" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb20-106"><a href="#cb20-106" aria-hidden="true" tabindex="-1"></a>  all_res_meta,</span>
<span id="cb20-107"><a href="#cb20-107" aria-hidden="true" tabindex="-1"></a>  diff_res <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">id =</span> ids)</span>
<span id="cb20-108"><a href="#cb20-108" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-109"><a href="#cb20-109" aria-hidden="true" tabindex="-1"></a>all_res_meta<span class="sc">$</span>exp_diff_fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(all_res_meta<span class="sc">$</span>p_value,<span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb20-110"><a href="#cb20-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-111"><a href="#cb20-111" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_res_meta, <span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/filter_target0331.rds&quot;</span>)</span></code></pre></div>
<p>For each gene we define differential expression statistics and
perturbation statistics：Differential Expression Stat =
(-log10(Exp-FDR)) * Exp_FC; Perturbation Stat =
(-log10(Perturbation-FDR)) * Abs_FC, and sort the genes according to
these two statistics.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/filter_target0331.rds&quot;</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>tt <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(exp_diff_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> exp_fc <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_pre_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(Goal_end_FDR)) <span class="sc">*</span> abs_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exp_pre_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(exp_diff_fdr)) <span class="sc">*</span> exp_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_stat =</span> <span class="fu">log10</span>(pertur_pre_stat),</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">exp_stat =</span> <span class="fu">log10</span>(exp_pre_stat)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(pertur_stat),<span class="fu">desc</span>(exp_stat))</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtext)</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> tt <span class="sc">%&gt;%</span> <span class="fu">select</span>(Gene_name,exp_stat,pertur_stat) <span class="sc">%&gt;%</span> </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Differential Expression Stat</span><span class="st">`</span><span class="ot">=</span>exp_stat,</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Perturbation Stat</span><span class="st">`</span><span class="ot">=</span>pertur_stat) <span class="sc">%&gt;%</span> </span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;Differential Expression Stat&quot;</span>,</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;Perturbation Stat&quot;</span>),</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">names_to =</span> <span class="st">&quot;Score Type&quot;</span>,<span class="at">values_to =</span> <span class="st">&quot;score&quot;</span>)</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(dt,<span class="at">x=</span><span class="st">&quot;Gene_name&quot;</span>,<span class="at">y=</span><span class="st">&quot;score&quot;</span>,<span class="at">fill =</span> <span class="st">&quot;Score Type&quot;</span>,</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>),<span class="at">xlab =</span> F,<span class="at">ylab =</span> <span class="st">&quot;log10(Score)&quot;</span>,</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">lab.size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_markdown</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-19-1.png" width="2400" /></p>
<p>Finally, we obtained 110 metabolic genes that met the criterias. We
performed metabolic pathway over representation enrichment analysis on
these genes and found that the most significantly enriched metabolic
pathway was oxidative phosphorylation.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pathway,genes)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(kegg) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;gs_name&quot;</span>,<span class="st">&quot;input_id&quot;</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>enz_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(<span class="fu">unique</span>(kegg<span class="sc">$</span>input_id), <span class="at">transTo =</span> <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Some ID occurs one-to-many match, like &quot;A2M, AAAS, ABCB11&quot;...</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 99.81% genes are mapped to entrezid</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">left_join</span>(kegg,enz_id)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(input_id)`</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in left_join(kegg, enz_id): Detected an unexpected many-to-many relationship between `x` and `y`.</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Row 1 of `x` matches multiple rows in `y`.</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ Row 51 of `y` matches multiple rows in `x`.</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ℹ If a many-to-many relationship is expected, set `relationship =</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &quot;many-to-many&quot;` to silence this warning.</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gs_name,entrezid) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">entrez_gene =</span> entrezid)</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>enz_id <span class="ot">&lt;-</span> genekitr<span class="sc">::</span><span class="fu">transId</span>(<span class="fu">unique</span>(tt<span class="sc">$</span>Gene_name), <span class="at">transTo =</span> <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Some ID occurs one-to-many match, like &quot;B3GALNT2, CES1, CHURC1-FNTB&quot;...</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 100% genes are mapped to entrezid</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>em <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enricher</span>(enz_id<span class="sc">$</span>entrezid, <span class="at">TERM2GENE=</span>kegg,</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pvalueCutoff =</span> <span class="dv">1</span>, <span class="at">qvalueCutoff =</span><span class="dv">1</span>)<span class="do">##use all gene as background</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>em_res <span class="ot">&lt;-</span> em <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>em_res <span class="ot">&lt;-</span> em_res <span class="sc">%&gt;%</span> </span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; -.+&quot;</span>,<span class="st">&quot;&quot;</span>,ID)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pathway <span class="sc">%in%</span> kegg<span class="sc">$</span>pathway) <span class="sc">%&gt;%</span> </span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>em <span class="ot">&lt;-</span> <span class="fu">filter</span>(em, ID <span class="sc">%in%</span> em_res<span class="sc">$</span>ID)</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>em<span class="sc">@</span>result<span class="sc">$</span>Description <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; -.+&quot;</span>,<span class="st">&quot;&quot;</span>,em<span class="sc">@</span>result<span class="sc">$</span>Description)</span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(em, <span class="at">showCategory=</span><span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels=</span><span class="cf">function</span>(x) stringr<span class="sc">::</span><span class="fu">str_wrap</span>(x, <span class="at">width=</span><span class="dv">40</span>))</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Scale for y is already present.</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adding another scale for y, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-20-1.png" width="2400" /></p>
<p>Subsequently, we extracted genes that have protein interactions with
oxidative phosphorylation pathway genes from the STRING database to form
a protein-protein interaction network.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rbioapi)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>oxi <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">&quot;Oxidative phosphorylation&quot;</span>)</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>oxi <span class="ot">&lt;-</span> oxi <span class="sc">%&gt;%</span> <span class="fu">select</span>(genes)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>proteins <span class="ot">&lt;-</span> oxi<span class="sc">$</span>genes</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="do">## map our protein IDs</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>proteins_mapped <span class="ot">&lt;-</span> <span class="fu">rba_string_map_ids</span>(<span class="at">ids =</span> proteins, <span class="at">species =</span> <span class="dv">9606</span>)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>int_partners <span class="ot">&lt;-</span> <span class="fu">rba_string_interaction_partners</span>(<span class="at">ids =</span> proteins_mapped<span class="sc">$</span>stringId,</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">species =</span> <span class="dv">9606</span>,</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">required_score =</span> <span class="dv">700</span>)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(int_partners, <span class="at">file =</span> <span class="st">&quot;~/hypoxia_target/data/ppi_oxi.rds&quot;</span>)</span></code></pre></div>
<p>The network has a total of 413 gene nodes, of which 33 genes are in
our screening list. We extracted the smallest subgraph connecting these
33 gene nodes by minimum spanning tree based approximation (KB)
algorithm.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/filter_target0331.rds&quot;</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>ppi_oxi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/ppi_oxi.rds&quot;</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(exp_diff_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> exp_fc <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(Goal_end_FDR)) <span class="sc">*</span> <span class="fu">log10</span>(abs_fc)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exp_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(exp_diff_fdr)) <span class="sc">*</span> <span class="fu">log10</span>(exp_fc)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(pertur_stat),<span class="fu">desc</span>(exp_stat))</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> ppi_oxi <span class="sc">%&gt;%</span> </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(score <span class="sc">&gt;</span> <span class="fl">0.9</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(preferredName_A, preferredName_B)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>dt_in <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">genes =</span> <span class="fu">c</span>(<span class="fu">unique</span>(<span class="fu">c</span>(dt<span class="sc">$</span>preferredName_A,dt<span class="sc">$</span>preferredName_B)))</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Type =</span> <span class="fu">ifelse</span>(genes <span class="sc">%in%</span> all_res_meta<span class="sc">$</span>Gene_name,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>dt_in<span class="sc">$</span>Type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt_in<span class="sc">$</span>Type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggnetwork)</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Registered S3 method overwritten by &#39;ggnetwork&#39;:</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   method         from  </span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   fortify.igraph ggtree</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>graph <span class="ot">&lt;-</span> <span class="fu">simplify</span>(<span class="fu">graph_from_data_frame</span>(<span class="at">d =</span> dt, </span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">vertices =</span> dt_in, </span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">directed =</span> <span class="cn">FALSE</span>))</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> ggnetwork<span class="sc">::</span><span class="fu">ggnetwork</span>(graph, <span class="at">arrow.gap =</span> <span class="dv">0</span>, </span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>                          <span class="at">layout=</span> igraph<span class="sc">::</span><span class="fu">layout_with_fr</span>(graph))</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(n, <span class="fu">aes</span>(<span class="at">x =</span> .data<span class="sc">$</span>x, <span class="at">y =</span> .data<span class="sc">$</span>y, </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>              <span class="at">xend =</span> .data<span class="sc">$</span>xend, </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">yend =</span> .data<span class="sc">$</span>yend)) <span class="sc">+</span> </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_edges</span>(<span class="at">color =</span> <span class="st">&quot;grey75&quot;</span>,</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.5</span>, </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>             <span class="at">curvature =</span> <span class="dv">0</span>, </span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>             <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_nodes</span>(<span class="fu">aes</span>(<span class="at">color =</span> .data<span class="sc">$</span>Type)) <span class="sc">+</span> </span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>  ggnetwork<span class="sc">::</span><span class="fu">theme_blank</span>()</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-22-1.png" width="2400" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="do">####提取子图</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SteinerNet)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qgraph)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>ppi_g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(dt)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>in_genes <span class="ot">&lt;-</span> dt_in<span class="sc">$</span>genes[<span class="fu">which</span>(dt_in<span class="sc">$</span>Type <span class="sc">==</span> <span class="st">&quot;yes&quot;</span>)]</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">steinertree</span>(<span class="at">type =</span> <span class="st">&quot;KB&quot;</span>, <span class="at">terminals =</span> in_genes,</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">graph =</span> ppi_g, <span class="at">color =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>out_dt <span class="ot">&lt;-</span> <span class="fu">as_data_frame</span>(out[[<span class="dv">1</span>]])</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="do">##plot</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>out_g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(out_dt,<span class="at">directed =</span> F)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/meta_target/data/kegg_all_pathway.rds&quot;</span>)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> </span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Metabolism&quot;</span>,class) <span class="sc">|</span> <span class="fu">grepl</span>(<span class="st">&quot;metabolism&quot;</span>,pathway)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">- Homo sapiens </span><span class="sc">\\</span><span class="st">(human</span><span class="sc">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,pathway))</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>oxi <span class="ot">&lt;-</span> kegg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">&quot;Oxidative phosphorylation&quot;</span>)</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>oxi <span class="ot">&lt;-</span> oxi <span class="sc">%&gt;%</span> <span class="fu">select</span>(genes)</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>oxi_gene <span class="ot">&lt;-</span> in_genes[<span class="fu">which</span>((in_genes <span class="sc">%in%</span> oxi<span class="sc">$</span>genes))]</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>not_oxi_gene <span class="ot">&lt;-</span> in_genes[<span class="fu">which</span>(<span class="sc">!</span>(in_genes <span class="sc">%in%</span> oxi<span class="sc">$</span>genes))]</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(out_g)[oxi_gene]<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="st">&quot;#11325D&quot;</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(out_g)[not_oxi_gene]<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="st">&quot;#F5A673&quot;</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>e <span class="ot">&lt;-</span> <span class="fu">get.edgelist</span>(out_g,<span class="at">names=</span><span class="cn">FALSE</span>)</span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>l <span class="ot">&lt;-</span> <span class="fu">qgraph.layout.fruchtermanreingold</span>(e,<span class="at">vcount=</span><span class="fu">vcount</span>(out_g))</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)<span class="sc">+</span>.<span class="dv">1</span>)</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_g,<span class="at">vertex.size=</span><span class="dv">8</span>, <span class="at">edge.arrow.size=</span><span class="fl">0.3</span>,</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label =</span> <span class="fu">V</span>(out_g)<span class="sc">$</span>name,</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label.dist=</span><span class="fl">1.5</span>,<span class="at">layout=</span>l, <span class="at">vertex.label.cex =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-22-2.png" width="2400" /></p>
<p>We found that there are two genes in this subgraph that are not genes
in the oxidative phosphorylation pathway (FLAD1 and SUCLG1). Since
oxidative phosphorylation is also important in normal cells, these two
genes may be potential hypoxia targets.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/filter_target0331.rds&quot;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>tt <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Gene_name <span class="sc">%in%</span> in_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(exp_diff_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> exp_fc <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_pre_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(Goal_end_FDR)) <span class="sc">*</span> abs_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exp_pre_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(exp_diff_fdr)) <span class="sc">*</span> exp_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_stat =</span> <span class="fu">log10</span>(pertur_pre_stat),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">exp_stat =</span> <span class="fu">log10</span>(exp_pre_stat)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(pertur_stat),<span class="fu">desc</span>(exp_stat))</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> tt <span class="sc">%&gt;%</span> <span class="fu">select</span>(Gene_name,exp_stat,pertur_stat) <span class="sc">%&gt;%</span> </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Differential Expression Stat</span><span class="st">`</span><span class="ot">=</span>exp_stat,</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Perturbation Stat</span><span class="st">`</span><span class="ot">=</span>pertur_stat) <span class="sc">%&gt;%</span> </span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;Differential Expression Stat&quot;</span>,</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>                               <span class="st">&quot;Perturbation Stat&quot;</span>),</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">names_to =</span> <span class="st">&quot;Score Type&quot;</span>,<span class="at">values_to =</span> <span class="st">&quot;score&quot;</span>)</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>x_to_highlight <span class="ot">&lt;-</span> <span class="st">&quot;FLAD1&quot;</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(dt,<span class="at">x=</span><span class="st">&quot;Gene_name&quot;</span>,<span class="at">y=</span><span class="st">&quot;score&quot;</span>,<span class="at">fill =</span> <span class="st">&quot;Score Type&quot;</span>,</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>),<span class="at">xlab =</span> F,<span class="at">ylab =</span> <span class="st">&quot;log10(Score)&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_markdown</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="sc">~</span> <span class="fu">case_when</span>(</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>    .x <span class="sc">%in%</span> x_to_highlight <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">&quot;&lt;span style=&#39;color: red&#39;&gt;&lt;b&gt;&quot;</span>, .x, <span class="st">&quot;&lt;/b&gt;&lt;/span&gt;&quot;</span>),</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> .x</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>  ))</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-23-1.png" width="2400" /></p>
<p>We further compared the expression differences of these genes in
tumors and normal tissues:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>all_res_meta <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/hypoxia_target/data/filter_target0331.rds&quot;</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>tt <span class="ot">&lt;-</span> all_res_meta <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Gene_name <span class="sc">%in%</span> in_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(exp_diff_fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> exp_fc <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_pre_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(Goal_end_FDR)) <span class="sc">*</span> abs_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exp_pre_stat =</span> (<span class="sc">-</span><span class="fu">log10</span>(exp_diff_fdr)) <span class="sc">*</span> exp_fc) <span class="sc">%&gt;%</span> </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pertur_stat =</span> <span class="fu">log10</span>(pertur_pre_stat),</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">exp_stat =</span> <span class="fu">log10</span>(exp_pre_stat)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(pertur_stat),<span class="fu">desc</span>(exp_stat))</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>luad <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/TCGA-LUAD.htseq_fpkm-uq.tsv.gz&quot;</span>,</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data.table =</span> F)</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>luad <span class="ot">&lt;-</span> luad <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> tt<span class="sc">$</span>id)</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(luad) <span class="ot">&lt;-</span> luad<span class="sc">$</span>Ensembl_ID</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>luad<span class="sc">$</span>Ensembl_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>lusc <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;/home/data/sdb/wt/TCGA-LUSC.htseq_fpkm-uq.tsv.gz&quot;</span>,</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data.table =</span> F)</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>lusc <span class="ot">&lt;-</span> lusc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Ensembl_ID <span class="sc">%in%</span> tt<span class="sc">$</span>id)</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(lusc) <span class="ot">&lt;-</span> lusc<span class="sc">$</span>Ensembl_ID</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>lusc<span class="sc">$</span>Ensembl_ID <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>lung_exp <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(luad,lusc)</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>lung_exp <span class="ot">&lt;-</span> <span class="fu">t</span>(lung_exp) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>lung_exp<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(lung_exp)</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>lung_exp <span class="ot">&lt;-</span> lung_exp <span class="sc">%&gt;%</span> </span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_type =</span> <span class="fu">ifelse</span>(<span class="fu">as.numeric</span>(<span class="fu">substr</span>(sample,<span class="dv">14</span>,<span class="dv">15</span>)) <span class="sc">&lt;</span> <span class="dv">11</span>,<span class="st">&quot;cancer&quot;</span>,<span class="st">&quot;normal&quot;</span>))</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>lung_exp <span class="ot">&lt;-</span> lung_exp <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,sample_type,<span class="fu">everything</span>())</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>lung_exp <span class="ot">&lt;-</span> lung_exp <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(lung_exp),</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">names_to =</span> <span class="st">&quot;gene&quot;</span>,</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">values_to =</span> <span class="st">&quot;exp&quot;</span>)</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>lung_exp <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>  lung_exp,</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>  tt <span class="sc">%&gt;%</span> <span class="fu">select</span>(id,Gene_name) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene=</span>id)</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(gene)`</span></span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ids =</span> <span class="fu">unique</span>(lung_exp<span class="sc">$</span>Gene_name),</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>                       <span class="at">p_value =</span> <span class="cn">NA</span>)</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(diff_res)){</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> lung_exp <span class="sc">%&gt;%</span> </span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample_type,Gene_name,exp) <span class="sc">%&gt;%</span> </span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name <span class="sc">==</span> diff_res<span class="sc">$</span>ids[i])</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>sample_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>sample_type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;cancer&quot;</span>,<span class="st">&quot;normal&quot;</span>))</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>  dt_res <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(exp <span class="sc">~</span> sample_type, <span class="at">data=</span>dt)</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>  diff_res<span class="sc">$</span>p_value[i] <span class="ot">&lt;-</span> dt_res<span class="sc">$</span>p.value</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> diff_res <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p_value)</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>diff_res <span class="ot">&lt;-</span> diff_res <span class="sc">%&gt;%</span> </span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group1 =</span> <span class="st">&quot;cancer&quot;</span>, <span class="at">group2 =</span> <span class="st">&quot;normal&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Gene_name =</span> ids) <span class="sc">%&gt;%</span> </span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y.position =</span> <span class="dv">25</span>)</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>diff_res<span class="sc">$</span>p.adj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(diff_res<span class="sc">$</span>p_value,<span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>diff_res<span class="sc">$</span>p.adj <span class="ot">&lt;-</span> <span class="fu">format</span>(diff_res<span class="sc">$</span>p.adj,<span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>lung_exp<span class="sc">$</span>Gene_name <span class="ot">&lt;-</span> <span class="fu">factor</span>(lung_exp<span class="sc">$</span>Gene_name,<span class="at">levels =</span> diff_res<span class="sc">$</span>Gene_name)</span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>lung_exp<span class="sc">$</span>sample_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(lung_exp<span class="sc">$</span>sample_type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;normal&quot;</span>,<span class="st">&quot;cancer&quot;</span>))</span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggprism)</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(lung_exp, <span class="fu">aes</span>(<span class="at">x =</span> Gene_name, <span class="at">y =</span> exp)) <span class="sc">+</span></span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> sample_type)) <span class="sc">+</span></span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_pvalue</span>(diff_res, <span class="at">x =</span> <span class="st">&quot;Gene_name&quot;</span>)<span class="sc">+</span></span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>(<span class="dv">90</span>)<span class="sc">+</span></span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;log2(FPKM-UQ + 1)&quot;</span>)<span class="sc">+</span></span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-24-1.png" width="2400" /></p>
</div>
</div>

   
   
            
      

  <script>
    $(document).ready(function () {

			
 		
	    });
  </script>



    <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
	var script = document.createElement("script");
	script.type = "text/javascript";
	script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
	document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>
  
</body>
</html>
